File: src/main/java/com/qlik/demo/java/util/ModelMapper.java
package com.qlik.demo.java.util;

import com.qlik.demo.java.model.Booking;
import com.qlik.demo.java.model.BookingDocument;
import com.qlik.demo.java.model.Train;
import com.qlik.demo.java.model.TrainDocument;

public class ModelMapper {

    private ModelMapper() {
    }

    public static Train convert(final TrainDocument document) {
        return new Train(
                document.id(),
                document.destination(),
                document.seatOption()
        );
    }

    public static Booking convert(final BookingDocument document) {
        return new Booking(
                document.id(),
                document.userId(),
                document.destination(),
                document.bookingDate()
        );
    }
}

File: src/main/kotlin/com/qlik/demo/kotlin/util/ModelMapper.kt
package com.qlik.demo.kotlin.util

import com.qlik.demo.kotlin.model.Booking
import com.qlik.demo.kotlin.model.Train
import com.qlik.demo.kotlin.model.TrainDocument

object ModelMapper {
    fun convertToTrain(document: TrainDocument) =
        Train(
            id = document.id,
            destination = document.destination,
            seatOption = document.seatOption
        )

    fun convertToBooking(document: com.qlik.demo.kotlin.model.BookingDocument) =
        Booking(
            id = document.id,
            destination = document.destination,
            userId = document.userId,
            bookingDate = document.bookingDate
        )
}